atomic_reactor/core.py

```diff
@@ -471,14 +526,18 @@ def pull_image(self, image, insecure=False):
         logger.info("pulling image '%s' from registry", image)
         logger.debug("image = '%s', insecure = '%s'", image, insecure)
         try:
-            logs_gen = self.d.pull(image.to_str(tag=False), tag=image.tag,
-                                   insecure_registry=insecure, decode=True, stream=True)
+            command_result = self.retry_generator('pull_image', self.d.pull,
+                                                  image.to_str(tag=False),
+                                                  tag=image.tag, insecure_registry=insecure,
+                                                  decode=True, stream=True)
         except TypeError:
             # because changing api is fun
-            logs_gen = self.d.pull(image.to_str(tag=False), tag=image.tag, decode=True, stream=True)
-        command_result = wait_for_command(logs_gen)
+            command_result = self.retry_generator('pull_image', self.d.pull,
+                                                  image.to_str(tag=False),
+                                                  tag=image.tag, decode=True, stream=True)
+
         self.last_logs = command_result.logs
-        return image.to_str()
+        return command_result.parsed_logs
```

Comment:
Why is return value changed? Are we using pull logs somewhere?

Reply:
just to make it same as for push_image, and return value which is known even before call isn't very useful at all

Reply:
Well, we're not using `push_image` logs anywhere as well. Maybe it should return `None` for both then?

Reply:
yeah output of either isn't used anywhere

Reply:
what do you think @twaugh @lcarva  should we leave logs or None?

Reply:
Is leaving the return value the same as it was an option?

Reply:
yeah right I can leave it as it was